### DISCUSSION TITLE
linearization file

### DISCUSSION THREAD

**Discussion by joeb8799** [OpenFAST/openfast#1148] [General]
Hello,
I need to generate the linearization files (.lin) for a turbine model that i have in order to use them as an input for the ROSCO controller.
how can i generate these files, is there any documentation or guide on how to get the .lin files?

**Comment by jjonkman** [comment]
Dear @joeb8799,

Can you clarify what your intended use of the linearization files is?  Linearization has been discussed many times on this site and on our forum (https://forums.nrel.gov/) and documentation for various linearization applications is available.  But it would help to clarify what your use case is.

Best regards,

**Reply by joeb8799** [reply]
well I'm working on example 12 of the ROSCO Controller, in which robust scheduling is applied to get better control. The idea of the robust scheduling is that it takes the linearization files (.lin) as input. I just need to know how can I get these .lin files from OpenFAST, used in example 12, so that I can generate it for my turbine model and apply Robust scheduling to it.

**Reply by jjonkman** [reply]
Dear @joeb8799,

OK, I t looks like you asked a similar question on the ROSCO issues page (https://github.com/NREL/ROSCO/discussions/146), which referenced an OpenFAST issue that discusses linearization (https://github.com/OpenFAST/openfast/issues/738), including a reference to @ebranlard's proposed OpenFAST user documentation (https://ebranlard-openfast.readthedocs.io/en/doc/source/working.html).  Do you have a specific question based on these links?

Best regards,

**Comment by joeb8799** [comment]
Hello Mr,
I'm trying to generate the linearization files of the https://github.com/NREL/ROSCO/tree/main/Test_Cases/IEA-15-240-RWT-UMaineSemi. I have a question regarding the number of inputs in the .lin files, For my case, it is 160, I noticed that the 160 is 10 + 150 files related to the AeroDyn user property on blade 1 2 and 3. How can I disable this feature so that in the .lin files, only the 10 inputs are displayed?

**Reply by jjonkman** [reply]
Dear @joeb8799,

A similar question was asked and answered in the following OpenFAST issues: https://github.com/OpenFAST/openfast/issues/738 and https://github.com/OpenFAST/openfast/issues/778.

Best regards,
